packages/trader/src/App/Containers/trade-header-extensions.jsx

```diff
@@ -46,36 +49,38 @@ class TradeHeaderExtensions extends React.Component {
         populateHeaderExtensions(header_items);
     };
 
-    async componentDidMount() {
-        if (isMobile()) {
-            const { client } = this.props.store;
-            // Waits for login to complete
-            await when(() => !client.is_populating_account_list);
-            if (this.props.is_logged_in) {
-                await WS.wait('authorize');
-                this.props.onMountPositions();
-                this.props.onMountCashier(true);
-                this.props.setAccountSwitchListener();
+    React.useEffect(() => {
+        const waitForLogin = async () => {
+            if (isMobile()) {
+                const { client } = store;
+                // Waits for login to complete
+                await when(() => !client.is_populating_account_list);
+                if (is_logged_in) {
+                    await WS.wait('authorize');
+                    onMountPositions();
+                    onMountCashier(true);
+                    setAccountSwitchListener();
+                }
             }
-        }
 
-        this.populateHeader();
-    }
+            populateHeader();
+        };
+
+        waitForLogin();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
 
-    componentDidUpdate() {
-        this.populateHeader();
-    }
+    React.useEffect(() => {
+        populateHeader();
 
-    componentWillUnmount() {
-        if (isMobile()) this.props.onUnmountPositions();
-        this.props.populateHeaderExtensions(null);
-    }
+        return () => {
+            if (isMobile()) onUnmountPositions();
+            populateHeaderExtensions(null);
+        };
```

Comment:
this block will run after every update. I think we should move it to L70 after `waitForLogin` call so that it only runs on unmount

Reply:
Yes I'll move it